subscribe.js

```diff
@@ -0,0 +1,40 @@
+const silo = require('./combineNodes.js').silo;
+
+const subscribe = (component, name) => {
+    if(!name && !component.prototype){
+        throw Error('you cant use an anonymous function in subscribe without a name argument');
+    } else if (!name && !!component.prototype){
+        name = component.prototype.constructor.name + 'State'
+    }
+    
+    const searchSilo = (head, name) => {
+        
+        let children;
+        if(typeof head.value !== 'object') return null;
+        else children = head.value;
+        console.log('keys: ', Object.keys(children));
+
+        for(let i in children){
+          console.log(i, name);
+            if(i === name){
+                console.log('found it!');
+                return children[i]
+            } else {
+                let foundNode = searchSilo(children[i], name);
+                console.log('coming back up. foundNode is: ', foundNode);
+                if(!!foundNode){return foundNode};
+            }
+        }
+    }
+
+    let foundNode = searchSilo(silo, name);
+    foundNode._subscribers.push(component)
+    return foundNode;
```

Comment:
The foundNode shouldn't be returned, the subscribe function should return a function that will when called remove the subscribed function from the list, unsubscribing it.

Reply:
You're right. this was an artifact of my internal testing.